### ISSUE TITLE
[$250] Shortcut - Workspace avatar next to request shortcut does not update when it is changed

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#39045]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.57-0
**Reproducible in staging?:** y
**Reproducible in production?:** new feature
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to staging.new.expensify.com.
2. Create a new workspace and do not change workspace name and avatar.
3. Request money from workspace chat.
4. Click FAB and note the workspace avatar next to Request money shortcut.
5. Go to Profile > Workspaces > select workspace in Step 2.
6. Change workspace name and avatar.
7. Go to Chat tab.
8. Repeat Step 4.

## Expected Result:
The workspace avatar next to Request money shortcut will update.

## Actual Result:
The workspace avatar next to Request money shortcut remains as the default avatar.

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/71587b3b-796e-49f4-9fd9-3e866e7c4323

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e4ddf9604c71c3e3</li>
        <li>Upwork Job ID: 1773056493413158912</li>
        <li>Last Price Increase: 2024-03-27</li>
<li>Automatic offers: </li>
<ul>
<li>jjcoffee | Reviewer | 0</li>
<li>ZhenjaHorbach | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb

**Comment by amyevans** [comment]
This is in the Quick Action button new feature (cc @Gonals), it's a relatively minor UI bug so no need to block deploy.

**Comment by jjcoffee** [comment]
Reviewing Monday! :pray:

**Comment by jjcoffee** [comment]
Thanks for the [proposal](https://github.com/Expensify/App/issues/39045#issuecomment-2023741021) @ZhenjaHorbach! It looks good overall, but I just need a few clarifications.

> As alternative we can get the current policy using a function from our utils using quickActionReport.policyID

To me this sounds like the best approach. Ideally we don't want to rerun the `useMemo` on _any_ policy change, only the policy linked to the `quickActionReport`. Have you tested this approach? And could you update your proposal to indicate it as the key approach.

> Plus for more correct results we can pass policy for ReportUtils.getIcon

I'm not sure what "more correct results" means in this context, can you expand? As far as I can tell `ReportUtils.getIcon` will get the correct icon as long as we rerun the `useMemo`.

**Comment by jjcoffee** [comment]
Sounds good, thanks! Happy to go with @ZhenjaHorbach's updated [proposal](https://github.com/Expensify/App/issues/39045#issuecomment-2023741021).

:ribbon::eyes::ribbon: C+ reviewed

**Comment by laurenreidexpensify** [comment]
Deployed to prod on 10 April, so should be eligible for payment on 17 April

**Comment by laurenreidexpensify** [comment]
Payment Summary:

C+ @jjcoffee $250 payment issued via upwork
Contributor @ZhenjaHorbach payment issued via upwork

**Comment by laurenreidexpensify** [comment]
@jjcoffee pls confirm if there are any regression / checklist steps to complete before we can close out, thanks

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/38669/
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/38669/files#r1568799896
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A - just missed in testing
- Determine if we should create a regression test for this bug. **Yes - if not already covered**
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. Create a new workspace
2. Request money from workspace chat.
3. Click FAB and note the workspace avatar next to Request money shortcut.
4. Go to Profile > Workspaces > select workspace in Step 2.
5. Change workspace name and avatar.
6. Go back to Chat tab.
7. Open the FAB and verify that the workspace name and avatar have changed in the shortcut

Do we agree 👍 or 👎
